sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidJoinRule.java

```diff
@@ -283,7 +283,8 @@ static boolean canHandleCondition(final RexNode condition, final RelDataType lef
 
   private static boolean isLeftExpression(final RexNode rexNode, final int numLeftFields)
   {
-    return ImmutableBitSet.range(numLeftFields).contains(RelOptUtil.InputFinder.bits(rexNode));
+    ImmutableBitSet inputBitSet = RelOptUtil.InputFinder.bits(rexNode);
+    return !inputBitSet.isEmpty() && ImmutableBitSet.range(numLeftFields).contains(inputBitSet);
```

Comment:
might be worth leaving a comment about why the bitset shouldn't be empty

Reply:
I added the empty check since without this (with the new changes), the `testFilterAndGroupByLookupUsingJoinOperator` was failing with `org.apache.druid.java.util.common.IAE: Cannot join lookup with condition referring to non-key column: (("dim2" == "j0.k") && ('xa' == "j0.v"))
	at org.apache.druid.segment.join.lookup.LookupJoinMatcher.create(LookupJoinMatcher.java:201)`.
The reason was that the left join in the query was getting converted to inner join with condition `foo.dim2 = lookyloo.k && lookyloo.v = 'xa'` which the lookup join matcher can't handle.
But now when I think of it more, I think this could generate sup-optimal plans for the IndexedTableMatcher which doesn't have such strict checks on the equi conditions.

Reply:
Also, this new check passes all the tests but fails to compile the query `SELECT dim1, l1.k FROM foo INNER JOIN (select k || '' as k from lookup.lookyloo group by 1) l1 ON foo.dim1 = l1.k and l1.k = 'abc'` 
will think more about fixing the original issue without modifying the condition strictness in equi-conditions.